npm run lint not recursing

1. OS. Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)
Fedora 23

2. Versions. Please run `ng --version`. If there's nothing outputted, please
[nbuso@linuxf]$ ng --version
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/user-guide/#watchman for more info.
angular-cli: 0.0.28
node: 4.2.4
os: linux x64
[nbuso@linuxf]$ npm --version
2.14.12

3. Repro steps. Tell us exactly what you did and how you came to the failure. 
From the command line run:
npm run lint   

4. The log given by the failure. Normally this include a stack trace and some
   more information.
it's supposed to recurse on subfolders but it stops on src/app/*.ts


5. Mention any other details that might be useful.
changing package.json in the script section to (added single quotes on file pattern):
    "lint": "tslint 'src/**/*.ts'",

it's working as expected

Heya!

I'm getting different behavior on windows, and the fix you propose actually gets me the wrong behavior.

On a newly generated project, I get:

```
filipe.silva@DESKTOP-7ND6T3R MINGW64 /E/dev/ideablade/testing/testproj5 (master)
$ npm run lint

> testproj5@0.0.0 lint E:\dev\ideablade\testing\testproj5
> tslint src/**/*.ts

src/app/route-config.ts[7, 3]: file should end with a newline
src/app/route-config.ts[5, 14]: variable name must be in camelcase or uppercase

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "lint"
npm ERR! node v4.2.1
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE
npm ERR! testproj5@0.0.0 lint: `tslint src/**/*.ts`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the testproj5@0.0.0 lint script 'tslint src/**/*.ts'.
npm ERR! This is most likely a problem with the testproj5 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tslint src/**/*.ts
npm ERR! You can get their info via:
npm ERR!     npm owner ls testproj5
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     E:\dev\ideablade\testing\testproj5\npm-debug.log
```

If I copy `src/app/route-config.ts` (which shows lint errors) one folder above and below, I get:

```
filipe.silva@DESKTOP-7ND6T3R MINGW64 /E/dev/ideablade/testing/testproj5 (master)
$ npm run lint

> testproj5@0.0.0 lint E:\dev\ideablade\testing\testproj5
> tslint src/**/*.ts

src/app/app2/route-config.ts[7, 3]: file should end with a newline
src/app/app2/route-config.ts[5, 14]: variable name must be in camelcase or uppercase
src/app/route-config.ts[7, 3]: file should end with a newline
src/app/route-config.ts[5, 14]: variable name must be in camelcase or uppercase
src/route-config.ts[7, 3]: file should end with a newline
src/route-config.ts[5, 14]: variable name must be in camelcase or uppercase

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "lint"
npm ERR! node v4.2.1
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE
npm ERR! testproj5@0.0.0 lint: `tslint src/**/*.ts`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the testproj5@0.0.0 lint script 'tslint src/**/*.ts'.
npm ERR! This is most likely a problem with the testproj5 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tslint src/**/*.ts
npm ERR! You can get their info via:
npm ERR!     npm owner ls testproj5
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     E:\dev\ideablade\testing\testproj5\npm-debug.log
```

Applying your fix then gets me:

```
filipe.silva@DESKTOP-7ND6T3R MINGW64 /E/dev/ideablade/testing/testproj5 (master)
$ npm run lint

> testproj5@0.0.0 lint E:\dev\ideablade\testing\testproj5
> tslint 'src/**/*.ts'
```

This might be a different implementation in linux and windows machines though, since the windows console doesn't expand double stars (and thus many packages use `node-glob` to do it).

@angular/angular-cli-team anyone can help me debug this?

--- a/addon/ng2/blueprints/ng2/files/package.json
+++ b/addon/ng2/blueprints/ng2/files/package.json
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "ng server",
     "postinstall": "typings install --ambient",
-    "lint": "tslint src/**/*.ts",
+    "lint": "tslint \"src/**/*.ts\"",
     "format": "clang-format -i -style=file --glob=src/**/*.ts",
     "pree2e": "webdriver-manager update",
     "e2e": "protractor"

